atoma-daemon/src/daemon.rs

```diff
@@ -0,0 +1,328 @@
+    tcp_listener: TcpListener,
+    mut shutdown_receiver: Receiver<bool>,
+) -> anyhow::Result<()> {
+    let daemon_router = create_daemon_router(daemon_state);
+    let server = axum::serve(tcp_listener, daemon_router.into_make_service())
+        .with_graceful_shutdown(async move {
+            shutdown_receiver
+                .changed()
+                .await
+                .expect("Error receiving shutdown signal")
+        });
+    server.await?;
+    Ok(())
+}
+
+/// Creates and configures the main router for the Atoma daemon HTTP API.
+///
+/// # Arguments
+/// * `daemon_state` - The shared state container that will be available to all route handlers
+///
+/// # Returns
+/// * `Router` - A configured axum Router instance with all API routes and shared state
+///
+/// # API Endpoints
+///
+/// ## Subscription Management
+/// * `GET /subscriptions` - Get all subscriptions for registered nodes
+/// * `GET /subscriptions/:id` - Get subscriptions for a specific node
+/// * `POST /model_subscribe` - Subscribe a node to a model
+/// * `POST /task_subscribe` - Subscribe a node to a task
+/// * `POST /task_update_subscription` - Updates an already existing subscription to a task
+/// * `POST /task_unsubscribe` - Unsubscribe a node from a task
+///
+/// ## Task Management
+/// * `GET /tasks` - Get all available tasks
+///
+/// ## Stack Operations
+/// * `GET /stacks` - Get all stacks for registered nodes
+/// * `GET /stacks/:id` - Get stacks for a specific node
+/// * `GET /almost_filled_stacks/:fraction` - Get stacks filled above specified fraction
+/// * `GET /almost_filled_stacks/:id/:fraction` - Get node's stacks filled above fraction
+/// * `GET /claimed_stacks` - Get all claimed stacks
+/// * `GET /claimed_stacks/:id` - Get claimed stacks for a specific node
+/// * `POST /try_settle_stack_ids` - Attempt to settle specified stacks
+/// * `POST /submit_stack_settlement_attestations` - Submit attestations for stack settlement
+/// * `POST /claim_funds` - Claim funds from completed stacks
+///
+/// ## Attestation Disputes
+/// * `GET /against_attestation_disputes` - Get disputes against registered nodes
+/// * `GET /against_attestation_disputes/:id` - Get disputes against a specific node
+/// * `GET /own_attestation_disputes` - Get disputes initiated by registered nodes
+/// * `GET /own_attestation_disputes/:id` - Get disputes initiated by a specific node
+///
+/// ## Node Registration
+/// * `POST /register` - Register a new node
+///
+/// # Example
+/// ```rust,ignore
+/// use atoma_daemon::DaemonState;
+///
+/// let daemon_state = DaemonState::new(/* ... */);
+/// let app = create_daemon_router(daemon_state);
+/// // Start the server with the configured router
+/// axum::Server::bind(&"0.0.0.0:3000".parse().unwrap())
+///     .serve(app.into_make_service())
+///     .await?;
+/// ```
+pub fn create_daemon_router(daemon_state: DaemonState) -> Router {
+    Router::new()
+        .route("/subscriptions/:id", get(get_node_subscriptions))
```

Comment:
add a `subscriptions` endpoint too. Do we really want a subscription by id ?

Reply:
changed to all subscriptions